api/routes/v1.js

```diff
@@ -0,0 +1,349 @@
+        console.error(`REALERROR:/${error}`)
+        if(error.name === 'CastError'){
+            const e = new Error("There is no unit with that number")
+            e.status = 404
+            next(e)
+        }
+        if(error.name === 'TypeError'){
+            const e = new Error("There is no comapany for this unit")
+            e.status = 404
+            next(e)
+        }
+        if(error.name === 'ValidationError'){
+            const e = new Error(error.message)
+            e.status = 400
+            next(e)
+        }
+        const e = new Error(error.message)
+        e.status = 400
+        next(e)
+    }
+})
+
+// PATCH /api/v1/units/[id]/company/employees/[id]
+//Update an employee and return that employee for the given company. If no company is listed, return a 404
+//and an appropriate message. If the unit ID provided does not match a unit,
+//return a 404 and a _different_ appropriate message. If the employee information is malformed in any way,
+// return a 400 and an error message with as much detail as possible. If no employee with that ID exists,
+// return a _different_ appropriate message.
+router.patch('/units/:id/company/employees/:eid', async (req, res, next)=> {
+    let status = 201
+    try{
+        const unit = await Units.findOne({_id: req.params.id})
+        const theCompany = unit.company
+        const employee = theCompany.employees.id({_id: req.params.eid})
+        employee.set(req.body)
+        await unit.save()
+        const response = employee
+        res.json({status, response})
+    }catch(error){
+        console.error(`REALERROR:/${error}`)
+        if(error.name === 'CastError'){
+            const e = new Error("There is no unit with that number")
+            e.status = 404
+            next(e)
+        }
+        if(error.name === 'TypeError'){
+            const e = new Error("There is no employee with that number")
+            e.status = 404
+            next(e)
+        }
+        if(error.name === 'ValidationError'){
+            const e = new Error(error.message)
+            e.status = 400
+            next(e)
+        }
+        const e = new Error("Something went wrong")
+        e.status = 400
+        next(e)
+    }
+})
+
+/// DELETE /api/v1/units/[id]/company/employees/[id]
+//Destroy the employee document and return that employee's document for the given company.
+//If no company is listed, return a 404 and an appropriate message.
+//If the unit ID provided does not match a unit, return a 404 and a _different_ appropriate message.
+//If no employee with that ID exists, return a _different_ appropriate message.
+router.delete('/units/:id/company/employees/:eid', async (req, res, next)=> {
+    const status = 201
+    try{
+        const unit = await Units.findOne({_id: req.params.id})
+        const theCompany = unit.company
+        const employee = theCompany.employees.id({_id: req.params.eid})
+        response = employee
+        employee.remove()
+        unit.save()
```

Comment:
I think you'll want an `await` here to ensure that the `.save()` finishes before responding.